conn_pool.go

```diff
@@ -335,8 +398,16 @@ func (p *ConnPool) Prepare(name, sql string) (*PreparedStatement, error) {
 	if err != nil {
 		return nil, err
 	}
-	ps, err := c.Prepare(name, sql)
 	p.availableConnections = append(p.availableConnections, c)
+
```

Comment:
Shouldn't we also be signaling when putting the connection back to `availableConnections`?